libqtile/widget/net.py

```diff
@@ -41,7 +41,7 @@ class Net(base.ThreadPoolText):
     defaults = [
         (
             "format",
-            "{interface}: {down} \u2193\u2191 {up}",
+            "{interface}: {down}{down_suffix} \u2193\u2191 {up}{up_suffix}",
```

Comment:
You need to add some default precision for `down` and `up` as, otherwise, you can get really long floats with 5 or 6 decimal places.

The value you set shouldn't cause the tests to fail.

Reply:
I've rounded the default to two decimal places, so that the default format doesn't actually change from what it is now.
Is that okay?

Reply:
Does that mean that uses of the widget won't have anything change for them at all? because that would be ideal

Reply:
> Does that mean that users of the widget won't have anything change for them at all? because that would be ideal

Yeah, except they can now change it if they want to.

Reply:
Perfect!

Reply:
> Does that mean that users of the widget won't have anything change for them at all? because that would be ideal

Not *quite*...

The previous version was worded as it was as it fixed the length of the `up` and `down` text to stop the widget resizing, even if the units changed.

That's not being done here. To do it, we'd need to do two things:
- make the speed a fixed length e.g. `{down:5.2f}`
- make the units a fixed length e.g. `{down_unit:<2}` (left aligning)

Reply:
https://user-images.githubusercontent.com/89494022/162620104-983a4532-ce94-41df-b963-2ae5eca6166c.mp4

I don't think it worked like that :thinking:. Your suggested changes also didn't do anything. (The video is from the master branch.)

Reply:
Maybe I misremembered. Will check when I'm back home.

Reply:
Looking at the code, I'm sure I am right. I think the issue is in the example I gave you. 5 was too few characters as `100.00` is 6 characters. So try `{down:7.2f}`

Reply:
Nope, doesn't work.

Reply:
Ah. It will only fix the width if you're using a monospace font...

Reply:
My bad.

Reply:
> My bad.

Not at all.

Monospace fonts aren't the default so most people are going to see changing width. However, latest change keeps this consistent with previous defaults.

Reply:
Done